pkg/config/deployment.go

```diff
@@ -523,3 +531,76 @@ func (n *NotificationMention) Validate() error {
 	}
 	return fmt.Errorf("event %q is incorrect as NotificationEventType", n.Event)
 }
+
+// PostSync provides all configurations to be used once the current deployment
+// is triggered successfully.
+type PostSync struct {
+	DeploymentChain *DeploymentChain `json:"chain"`
+}
+
+func (p *PostSync) Validate() error {
+	if dc := p.DeploymentChain; dc != nil {
+		return dc.Validate()
+	}
+	return nil
+}
+
+// DeploymentChain provides all configurations to used to trigger a chain of deployments.
+type DeploymentChain struct {
+	// Nodes provides list of DeploymentChainNodes which contain filters to be used
+	// to find applications to deploy as chain node. It's required to not empty.
+	Nodes []*DeploymentChainNode `json:"applications"`
+	// Conditions provides configuration used to determine should the piped in charge in
+	// the first applications in the chain trigger a whole new deployment chain or not.
+	// If this field is not set, always trigger a whole new deployment chain when the current
+	// application is triggered.
+	Conditions *DeploymentChainTriggerCondition `json:"conditions,omitempty"`
```

Comment:
"conditions" -> "condition"?

Reply:
I think it should be `conditions` since we may add more conditions under this section later 🙏

Reply:
Ok. I think we still don't know the exact use case for this, so how about commenting out this field for now.

Reply:
sure 👍

Reply:
addressed by https://github.com/pipe-cd/pipe/pull/2792/commits/493d66bb5b0f8f2f3fb0058f36124719798fdf88